apollo-compiler/src/test/graphql/com/example/fragment_used_twice/TestQuery.kt

```diff
@@ -141,43 +152,154 @@ class TestQuery : Query<TestQuery.Data, Operation.Variables> {
         override val name: String
 
         override fun marshaller(): ResponseFieldMarshaller
+
+        interface Character : Human, HumanDetail.Character {
+          override val __typename: String
+
+          /**
+           * What this human calls themselves
+           */
+          override val name: String
+
+          /**
+           * The date character was born.
+           */
+          override val birthDate: Any
+
+          override fun marshaller(): ResponseFieldMarshaller
```

Comment:
Shouldn't we put the `marshaller()` on the class rather than the interface?

Reply:
Hm, user will mostly work with interfaces when it comes to fragments, so we need to provide access to marshaller via interface rather than via fragment implementation that can be intersection of fragments.

Reply:
I'm not sure how many users use `marshaller()` directly. My assumption was that very few users need to write  data/fragments and those who do use the imperative store APIs instead of calling `marshaller` directly.

More philosophically, what does marshalling an interface mean? There has to be a matching implementation to marshal something which is why it feels weird to have the method on the interface.

Reply:
Just to make sure we are on the same page, you want to remove `marshaller ` from the generated nested interfaces and leave it only for top level generated operation data classes:  `Operation.Data#responseFieldMapper()` and for `GraphqlFragment#marshaller`, right?

Reply:
The initial thought was to move them from the interfaces to the data classes but I think it makes sense to remove them from nested interfaces/classes altogether and only keep them in `Operation.Data#responseFieldMapper()` and `GraphqlFragment#marshaller`. Thinking more about this, maybe we could even remove `marshaller` from the public API and expose `Operation.Data#responseAdapter()` instead?

Reply:
Yeah, that makes sense, will do that in next PR